src/serve/grpc/query.rs

```diff
@@ -363,4 +398,59 @@ impl u5c::query::query_service_server::QueryService for QueryServiceImpl {
             next_token: String::default(),
         }))
     }
+
+    async fn read_chain_config(
+        &self,
+        request: Request<u5c::query::ReadChainConfigRequest>,
+    ) -> Result<Response<u5c::query::ReadChainConfigResponse>, Status> {
+        let message = request.into_inner();
+
+        info!("received new grpc query");
+
+        let tip = self.ledger.cursor()?;
+
+        let updates = self
+            .ledger
+            .get_pparams(tip.as_ref().map(|p| p.0).unwrap_or_default())?;
+
+        let updates: Vec<_> = updates
+            .into_iter()
+            .map(TryInto::try_into)
+            .try_collect::<_, _, pallas::codec::minicbor::decode::Error>()
+            .map_err(|e| Status::internal(e.to_string()))?;
+
+        let summary = pparams::fold_with_hacks(&self.genesis, &updates, tip.as_ref().unwrap().0);
+
+        let era_summaries = summary.past;
+        let era_summaries: Vec<_> = era_summaries
+            .into_iter()
+            .map(into_u5c_era_summary)
+            .collect();
+
+        let network_id = match self.genesis.shelley.network_id.as_ref() {
+            Some(network) => network.clone(),
+            None => "".to_string(),
+        };
+        let network_magic = self.genesis.shelley.network_magic.unwrap();
+        let caip2 = format!("cip34:{}-{}", network_id, network_magic);
+
+        let genesis = "".to_string();
```

Comment:
use the genesis block hash. For example, on mainnet, I believe the genesis block hash is `5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb`